subsys/usb/usb_transfer.c

```diff
@@ -313,8 +313,22 @@ int usb_transfer_sync(uint8_t ep, uint8_t *data, size_t dlen, unsigned int flags
 		return ret;
 	}
 
-	/* Semaphore will be released by the transfer completion callback */
-	k_sem_take(&pdata.sem, K_FOREVER);
+	/* Semaphore will be released by the transfer completion callback
+	 * which might not be called when transfer was cancelled
+	 */
+	while (1) {
+		struct usb_transfer_data *trans;
+
+		ret = k_sem_take(&pdata.sem, K_MSEC(100));
+		if (ret == 0)
+			break;
+
+		trans = usb_ep_get_transfer(ep);
+		if (!trans || trans->status != -EBUSY) {
+			LOG_DBG("Sync transfer cancelled, ep 0x%02x", ep);
```

Comment:
Does a bit higher LOG level make sense here, LOG_INF/LOG_WRN?

Reply:
I'm not sure if it is necessary to warn here more. I think that the case with cancelled transfer is quite common and valid.

Reply:
It may be valid but not common and should be treated accordingly by the application. At least LOG_INF would be appropriate to make the developer aware, LOG_DBG is too noisy.

Reply:
ok, done